xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -439,7 +439,7 @@ bool CDVDDemuxFFmpeg::Open(std::shared_ptr<CDVDInputStream> pInput, bool streami
   bool isBluray = pInput->IsStreamType(DVDSTREAM_TYPE_BLURAY);
   if (iformat && (strcmp(iformat->name, "mpegts") == 0) && !fileinfo && !isBluray)
   {
-    av_opt_set_int(m_pFormatContext, "analyzeduration", 500000, 0);
+    av_opt_set_int(m_pFormatContext, "analyzeduration", 2000000, 0);
```

Comment:
I would increase only for non-live streaming, see streaminfo, or it will increase channel switching times for some addons.

Reply:
I will do another patch later the day that will address the live streaming issue.

Reply:
If I understand correctly, this change will slow down opening of mpegts files (e.g. from LAN).

If you are using a wifi connection that is only just fast enough to stream these files (i.e. we can't read many times faster than realtime) then this could increase the half second delay to two seconds when starting playback.

Reply:
I've made some test with files, PVR recordings and PVR live TV. It seems that the changed line won't be called at all when watching live TV and switching channels. I'm testing with the TVHeadEnd backend. 
Increasing the analyzeduration FFMPEG value seems to be the only way to make the aac_latm files and PVR recording work and the 2 seconds is the lowest value in my tests that never let it fail.

I don't see the need to modify the patch, can you tell me the use case where it is a problem? Is this case so much more important than having at least a working player that can handle aac_latm audio stream files at all?

Reply:
I was referring to playback of a normal mpeg2ts file (not using PVR backend) - e.g. browsing through file menu.

Reply:
Yes...but the normal file needs this in case aac_latm stream is included, see here : https://github.com/xbmc/xbmc/issues/15087 or here : http://trac.kodi.tv/ticket/17927

Reply:
> It seems that the changed line won't be called at all when watching live TV and switching channels

it won't be called with tvheadend but tvheadend is not the only pvr addon.

Reply:
Okay, as I said soing a swtch case with the stream type is trivial...but at lease these two stream types need the increaed value:

DVDSTREAM_TYPE_FILE (Normal file)
DVDSTREAM_TYPE_PVRMANAGER (PVR Recording TVHeadEnd)

Is it okay to use the new value for these types? If yes, I would add another patch to this PR with the switch/case

Reply:
DVDSTREAM_TYPE_PVRMANAGER won't work because this can be live and recordings.

Reply:
Okay, and now? How to fix it? I'm out of options...is there any other way to differ in advance between working and non-working streams/files etc.? The Stream Type won't do it...

Reply:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L1579

change this to:
if (!mpegts)

Reply:
I've replaces the line you mentioned with this line 

if(strcmp(m_pFormatContext->iformat->name, "mpegts") != 0)

and it won't work after I reverted my patch

Reply:
So, what will you do? It seems that my patch not only fixes a lot of audio stream problems but also some video stream issues some users are experiencing. Wouldn't it be better to take the patch now and optimize later? Make it work, make it good, make it fast? We're now at "Make it work"...

Reply:
@FernetMenta Could you please tell us how we should proceed now to get this patch merged and/or to solve the problem (another way)?

Reply:
@nRaecheR you ask the wrong person. I am not team Kodi and don't decide to take or reject this change. I was asked for review and posted my opinion. This change increases channel switching times for pvr backends that don't have a demuxer. In addition @popcornmix raised a concern. 
I can help if somebody is interested in a fix that does not harm other use cases. Unfortunately I don't see much interest going into this direction. I just see users crying: "fix my use case, I don't care about the others."

Reply:
Yes, I'm aware that there seems to be some use cases where this patch may worsen the user experience. But I'm a normal user of Kodi/LibreElec and because I'm unable to use it with my setup (DVB-T2) since version 17.x I've done a lot of work to find the reason for this failure, created a patch and a pull request and my feeling is that no one from "Team Kodi" seems to care about the fix/my work/the problem at all....it's a little bit frustrating for a new contributor to the project.

Reply:
Sorry but we are not going to pull in a patch that fixes one problem and creates another. Regardless how much effort was put into finding the problem you had.
I will trust the opinion of the devs that know this part of the code.

Reply:
@nRaecheR fernet offered his help - we don't want to touch the analyzeduration and refrained from doing so already years ago.
So if @FernetMenta can help us getting a proper fix, I'd be interested.

Reply:
> my feeling is that no one from "Team Kodi" seems to care about the fix/my work/the problem at all....it's a little bit frustrating for a new contributor to the project.

Your feelings are understandable @nRaecheR and even though I am a team member I sometimes have the same, but I hope you will stick with it.  The active team is very small compared to the size of the project, and it could easily be that no one with sufficient knowledge in this area is available to review or take on the work.  Sorry I can't offer any input on videoplayer myself, I know nothing about that area of Kodi, but I would advise you to take on what @FernetMenta and @popcornmix have said.

@FernetMenta has said 
>I can help if somebody is interested in a fix that does not harm other use cases.

I suggest the best way to proceed is to take him up on that offer.

Reply:
Wonderful, as long as the bug will be fixed in one of the next stable releases of Leia I'm perfectly happy with any patch from anybody. I don't need my patch to be merged as long as the bug is finally fixed soon. I would be very glad if @FernetMenta will do another fix for the problem and will test it as soon as it is available.

Reply:
@nRaecheR he offered assistance and help, not fixing it for you.

Reply:
> @nRaecheR he offered assistance and help, not fixing it for you.

Haha, fixing it for me? Sorry, I've fixed it *for me*, it works perfectly *for me* with this proposed patch, now it's time to fix it *for you/Team Kodi*...sorry, that's out of my reach, as I said, I'm just a user of your software and not a member of the Team Kodi or the maintainer of the Videoplayer. 

I showed you where the problem is, I created a patch, it solves the problem. I think now it's time for the ones that rejected my patch to do a better one and finally make the software working.

Okay, I see, it seems to be difficult for an outsider to try to help here and others, I think I will stick with my custom build of LibreElec with my patch added to the patches queue for Kodi...